src/libraries/System.Text.Json/src/System/Text/Json/Serialization/Converters/Node/JsonNodeConverter.cs

```diff
@@ -16,9 +19,12 @@ internal sealed class JsonNodeConverter : JsonConverter<JsonNode?>
     {
         private static JsonNodeConverter? s_nodeConverter;
         private static JsonArrayConverter? s_arrayConverter;
+        private static ConcurrentDictionary<Type, JsonConverter>? s_jsonValuePrimitiveConverters;
```

Comment:
Do not store converters like this. Use a converter factory to create converter for each `JsonValuePrimitive<T>`. Check `NullableConverterFactory`.

Reply:
Thanks. Fixed by force push https://github.com/dotnet/runtime/pull/114863/commits/9047de2e4492706e2d91c31706344d4a6886fe4f